core/Updates/3.13.5-rc1.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Updates;
+
+use Piwik\DataAccess\ArchiveTableCreator;
+use Piwik\Date;
+use Piwik\Plugins\CoreAdminHome\Tasks\ArchivesToPurgeDistributedList;
+use Piwik\Updater;
+use Piwik\Updates as PiwikUpdates;
+
+class Updates_3_13_5_rc1 extends PiwikUpdates
+{
+    public function doUpdate(Updater $updater)
+    {
+        $this->addArchivesToPurge();
+    }
+
+    private function addArchivesToPurge()
+    {
+        $archivesToPurge = new ArchivesToPurgeDistributedList();
+
+        $startOfProblem = Date::factory('2020-03-01 00:00:00');
+
+        $archiveTables = ArchiveTableCreator::getTablesArchivesInstalled(ArchiveTableCreator::NUMERIC_TABLE);
+        foreach ($archiveTables as $table) {
+            $date = ArchiveTableCreator::getDateFromTableName($table);
+            list($year, $month) = explode('_', $date);
+
+            $dateObj = Date::factory("$year-$month-01 00:00:00");
+            if ($dateObj->isEarlier($startOfProblem)) { // only add if the table is for march 2020 or above since that is when the problem appeared
```

Comment:
I haven't checked it yet... but this would only run it for march if it was executed eg on April 23rd 2020 right? Although I suppose that's fine since we'd add today anyway on every purge and April would be deleted that way 👍

Reply:
I'm not sure what you mean, but it will run it if it finds an archive table for march or later. Though actually, this won't work because we also have to add the year table. I'll just add it if the year is 2020 or greater.

Reply:
I meant "2020-04-01 00:00:00" would be earlier than 2020-04-23 and therefore it wouldn't do it for April but it's not important since it would do it automatically anyway when the task is executed next

Reply:
The dates used should always be `-01 00:00:00` in this code, i'm not sure why the execution date of the update matters.